boilerplate/openshift/golang-osd-operator/rvmo-bundle.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+set -e
+
+REPOSITORY=${REPOSITORY:-"git@github.com:openshift/managed-release-bundle.git"}
+BRANCH=${BRANCH:-main}
+DELETE_TEMP_DIR=${DELETE_TEMP_DIR:-true}
+TMPD=$(mktemp -d --suffix -rvmo-bundle)
+[[ "${DELETE_TEMP_DIR}" == "true" ]] && trap 'rm -rf ${TMPD}' EXIT
+
+cd "${TMPD}"
+git clone --single-branch -b "${BRANCH}" "${REPOSITORY}" .
+bash hack/translate-operator-template.sh
```

Comment:
It seems this file doesn't exist on main branch which is the default, which will make it crash by default. 
Wouldn't it make more sense to error if no branch is specified (as I understand the git project isn't expected to have an actual implementation as I understand it)

Reply:
Operator being built from branch e.g. `release-4.15` needs to use the managed-release-bundle.git repo with the exact same branch. If that branch doesn't exist or there's anything wrong with the file being called, it needs to fail immediately, there should be no trying to recover from errors.

Reply:
resolving this based on conversation here https://redhat-internal.slack.com/archives/CFJD1NZFT/p1706703626887269?thread_ts=1706519071.058829&cid=CFJD1NZFT

Reply:
My point was not trying to recover, but to be explicit/provide guidance on the needed changes to have it work for user. Discussing with @jbpratt , the default to 'main' is only temp because branches on operators not being set yet so shouldn't be an issue by the time it is ready to be used/migrated to.